### PR TITLE
[#IOPID-1965]  Move io-functions-fast-login to ITN region

### PR DESCRIPTION
**Pull request by BurnedMarshal**
### List of changes
Add a new `fn-fast-login` Azure Function to the ITN Region

### Motivation and context
We are moving the fast-login Azure Function from WEU to ITN
The new DX module was used to create the new instance

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/domains/citizen-auth-app/07_function_fast_login.tf [hunk 1]
```diff
@@ -91,6 +92,72 @@ module "fast_login_snet" {
...
+    instance_number = "01"
+  }
+
+  resource_group_name = azurerm_resource_group.fast_login_rg_itn.name
+  health_check_path   = "/info"
+  node_version        = 18
+  tier                = "xl"
+
+  subnet_cidr                          = var.cidr_subnet_fnfastlogin_itn
+  subnet_pep_id                        = azurerm_subnet.fast_login_snet_itn.id
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/domains/citizen-auth-app/07_function_fast_login.tf]: Fixed in [158815d](https://github.com/pagopa/io-infra/pull/1197/commits/158815dc303fb0fb1af146926691540665332e2d)

File: src/domains/citizen-auth-app/07_function_fast_login.tf [hunk 2]
```diff
@@ -91,6 +92,140 @@ module "fast_login_snet" {
...
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "function_fast_login_itn" {
```
Comment (gquadrati) [inline src/domains/citizen-auth-app/07_function_fast_login.tf:120]: Autoscale settings are missing

File: src/domains/citizen-auth-app/01_network_itn.tf [hunk 3]
```diff
@@ -61,3 +61,48 @@ resource "azurerm_private_endpoint" "staging_function_lollipop_itn_sites" {
 
   tags = var.tags
 }
+
+
+## fn-fast-login-itn
+
+resource "azurerm_private_endpoint" "function_fast_login_itn_sites" {
```
Comment (gquadrati) [inline src/domains/citizen-auth-app/01_network_itn.tf:68]: I think it would be better to add a dependency to the function

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/domains/citizen-auth-app/01_network_itn.tf (hunk 3 above)
Comment (BurnedMarshal) [reply src/domains/citizen-auth-app/01_network_itn.tf:68]: What do you mean with dependency? Could be enough a comment that refers the function declaration?

File: src/domains/citizen-auth-app/07_function_fast_login.tf (hunk 2 above)
Comment (BurnedMarshal) [reply src/domains/citizen-auth-app/07_function_fast_login.tf:120]: Autoscale will be added with another PR https://github.com/pagopa/io-infra/pull/1198

**Review by gquadrati — APPROVED** [review]

**Pull request merged**